plugins/obs-webrtc/whip-output.cpp

```diff
@@ -323,21 +402,87 @@ bool WHIPOutput::Connect()
 	return true;
 }
 
-void WHIPOutput::StartThread()
+void WHIPOutput::SendOptions()
 {
-	if (!Setup())
+	if (endpoint_url.empty()) {
+		do_log(LOG_DEBUG,
+		       "No endpoint URL available, not sending OPTIONS");
 		return;
+	}
 
-	if (!Connect()) {
-		rtcDeletePeerConnection(peer_connection);
-		peer_connection = -1;
-		audio_track = -1;
-		video_track = -1;
-		return;
+	struct curl_slist *headers = NULL;
+	if (!bearer_token.empty()) {
+		auto bearer_token_header =
+			std::string("Authorization: Bearer ") + bearer_token;
+		headers =
+			curl_slist_append(headers, bearer_token_header.c_str());
 	}
 
-	obs_output_begin_data_capture(output, 0);
-	running = true;
+	std::string link_header;
+
+	// add user-agent to our requests
+	headers = curl_slist_append(headers, user_agent.c_str());
+
+	CURL *c = curl_easy_init();
+	curl_easy_setopt(c, CURLOPT_HEADERFUNCTION, curl_header_link_function);
+	curl_easy_setopt(c, CURLOPT_HEADERDATA, (void *)&link_header);
+	curl_easy_setopt(c, CURLOPT_HTTPHEADER, headers);
+	curl_easy_setopt(c, CURLOPT_URL, endpoint_url.c_str());
+	curl_easy_setopt(c, CURLOPT_CUSTOMREQUEST, "OPTIONS");
+	curl_easy_setopt(c, CURLOPT_TIMEOUT, 8L);
+
+	auto cleanup = [&]() {
+		curl_easy_cleanup(c);
+		curl_slist_free_all(headers);
+	};
+
+	CURLcode res = curl_easy_perform(c);
+	if (res == CURLE_OK) {
+		long response_code;
+		curl_easy_getinfo(c, CURLINFO_RESPONSE_CODE, &response_code);
+		// expected response per draft on OPTIONS is 200
+		if (response_code == 200) {
+			if (!link_header.empty()) {
+				do_log(LOG_INFO,
+				       "WHIP server provided a TURN server via the Link header");
+				struct curl_header *link_header_data;
+				CURLHcode h = curl_easy_header(
```

Comment:
there's an error here thrown in macOS clang because h is unused (warnings are turned into errors in obs).
So either just use the function without defining h if you're sure it can't fail, or if it's not a CURLHE_OK , do a cleanup.

Reply:
That's in the next push.